src/textual/_terminal_features.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+import os
+import platform
+from dataclasses import dataclass
+
+
+@dataclass
+class TerminalSupportedFeatures:
```

Comment:
This is the class that replaces the `this._sync_available: bool` we currently have on the App class.
I introduced because we now have 2 "sync" modes, each with their own set of ANSI sequences and their own machinery.
But also because we might have other optional terminal emulators features that we could use in the future, so adding flags on this class will likely scale better than adding them on the App class :slightly_smiling_face: